qemu/tests/qnum_change_reboot.py

```diff
@@ -0,0 +1,295 @@
+            if (cur_queues_status[1] != q_number
+                    or cur_queues_status[0] != queues_status[0]):
+                err_msg = "Param is valid, but change queues failed, "
+        elif cur_queues_status != queues_status:
+            if q_number != queues_status[1]:
+                err_msg = "Param is invalid, "
+            err_msg += "Current queues value is not expected, "
+            expect_q_number = queues_status[1]
+
+        if len(err_msg) > 0:
+            err_msg += "current queues set is %s, " % cur_queues_status[1]
+            err_msg += "max allow queues set is %s, " % cur_queues_status[0]
+            err_msg += "when run cmd: '%s', " % mq_set_cmd
+            err_msg += "expect queues are %s," % expect_q_number
+            err_msg += "expect max allow queues are %s, " % queues_status[0]
+            err_msg += "output: '%s'" % output
+            test.fail(err_msg)
+        return [int(_) for _ in cur_queues_status]
+
+    def get_queues_status(session, ifname, timeout=240):
+        """
+        Get queues status
+        """
+        mq_get_cmd = "ethtool -l %s" % ifname
+        nic_mq_info = session.cmd_output(mq_get_cmd, timeout=timeout, safe=True)
+        queues_reg = re.compile(r"Combined:\s+(\d)", re.I)
+        queues_info = queues_reg.findall(" ".join(nic_mq_info.splitlines()))
+        if len(queues_info) != 2:
+            err_msg = "Oops, get guest queues info failed, "
+            err_msg += "make sure your guest support MQ.\n"
+            err_msg += "Check cmd is: '%s', " % mq_get_cmd
+            err_msg += "Command output is: '%s'." % nic_mq_info
+            test.cancel(err_msg)
+        return [int(x) for x in queues_info]
+
+    def enable_multi_queues(vm):
+        sess = vm.wait_for_serial_login(timeout=login_timeout)
+        error_context.context("Enable multi queues in guest.", logging.info)
+        for nic_index, nic in enumerate(vm.virtnet):
+            ifname = utils_net.get_linux_ifname(sess, nic.mac)
+            queues = int(nic.queues)
+            change_queues_number(sess, ifname, queues)
+
+    def ping_test(dest_ip, ping_time, b_ping_lost_ratio, session=None):
+        status, output = utils_test.ping(dest=dest_ip, timeout=ping_time,
+                                         session=session)
+        packets_lost = utils_test.get_loss_ratio(output)
+        if packets_lost > b_ping_lost_ratio:
+            err = " %s%% packages lost during ping. " % packets_lost
+            err += "Ping command log:\n %s" % "\n".join(
+                output.splitlines()[-3:])
+            test.fail(err)
+
+    def test_netperf():
+        """
+        Netperf stress test for nic option.
+        """
+        netperf_client_link = os.path.join(data_dir.get_deps_dir("netperf"),
+                                           params.get("netperf_client_link"))
+        client_path = params.get("client_path")
+
+        n_client = utils_netperf.NetperfClient(
```

Comment:
We don't need to `port`, `prompt`, `linesep` and `status_test_command` for Linux guest.